tests/testthat/test-add_subprotocols.R

```diff
@@ -0,0 +1,179 @@
+    append = TRUE)
+
+  # add a sub-subprotocol to
+  # src/thematic/1_water/sfp-102-en_second-subprotocol
+  #debugonce(add_subprotocols)
+  index <- readLines(
+    con = "src/thematic/1_water/sfp-102-en_second-subprotocol/index.Rmd")
+  index[grepl("  dependencies_protocolcode:", index)] <-
+    "  dependencies_protocolcode: ['sfp-401-en']"
+  index[grepl("  dependencies_versionnumber:", index)] <-
+    "  dependencies_versionnumber: ['2020.02']"
+  index[grepl("  dependencies_params:", index)] <-
+    "  dependencies_params: ['list()']"
+  index[grepl("  dependencies_appendix:", index)] <-
+    "  dependencies_appendix: [TRUE]"
+  writeLines(
+    index,
+    con = "src/thematic/1_water/sfp-102-en_second-subprotocol/index.Rmd")
+
+  dependencies <- data.frame(
+    protocol_code = c("sfp-401-en"),
+    version_number = c("2020.02"),
+    params = c("list()"),
+    appendix = c(TRUE))
+
+  add_subprotocols(
+    .dependencies = dependencies,
+    fetch_remote = FALSE,
+    code_mainprotocol = 'sfp-102-en')
+
+
+  # add, commit and tag it
+  sfp_staged <- gert::git_add(files = ".")
+  gert::git_commit_all(message = "sfp-102-en_second-subprotocol")
+  specific_tag <- paste("sfp-102-en", version_number, sep = "-")
+  generic_tag <- paste("protocols", version_number, sep = "-")
+  gert::git_tag_create(name = specific_tag, message = "bla")
+  gert::git_tag_create(name = generic_tag, message = "bla")
+
+
+  # add subprotocols to
+  # src/project/mne/spp-001-en_mne-protocol/
+  #debugonce(add_subprotocols)
+  index <- readLines(
+    con = "src/project/mne/spp-001-en_mne-protocol/index.Rmd")
+  index[grepl("  dependencies_protocolcode:", index)] <-
+    "  dependencies_protocolcode: ['sfp-101-en', 'sfp-102-en']"
+  index[grepl("  dependencies_versionnumber:", index)] <-
+    "  dependencies_versionnumber: ['2020.01', '2020.04']"
+  index[grepl("  dependencies_params:", index)] <-
+    "  dependencies_params: ['list()', 'list(reviewers = c(\"reviewer1\", \"reviewer2\"))']"
+  index[grepl("  dependencies_appendix:", index)] <-
+    "  dependencies_appendix: [TRUE, TRUE]"
+  writeLines(
+    index,
+    con = "src/project/mne/spp-001-en_mne-protocol/index.Rmd")
+  dependencies <- data.frame(
+    protocol_code = c("sfp-101-en", "sfp-102-en"),
+    version_number = c("2020.01","2020.04"),
+    params = c("list()", "list(reviewers = c('reviewer1', 'reviewer2'))"),
+    appendix = c(TRUE, TRUE))
+  add_subprotocols(
+    .dependencies = dependencies,
+    fetch_remote = FALSE,
+    code_mainprotocol = 'spp-001-en')
+
+  # render spp-001-en including the subprotocol
+  render_protocol(protocol_code = "spp-001-en",
+                  params = list(
+                    dependencies_protocolcode = dependencies$protocol_code,
+                    dependencies_versionnumber = dependencies$version_number,
+                    dependencies_params = dependencies$params,
+                    dependencies_appendix = dependencies$appendix
```

Comment:
I may be mistaken, but why explicitly passing the `dependencies` params when rendering the _main_ protocol? The dependencies should only be defined in `index.Rmd` and are not meant to be set to another value in render - contrary to some regular params in the subprotocols.

Reply:
You are right. This can now be removed because I've added code that sets the correct values directly in the index.Rmd of the main protocol. Before, I had not done that and passing it via params was just a shortcut for testing purposes.